File: android/app/src/main/java/com/djs66256/short_drama/feature/earn/model/EarnTaskContext.kt
package com.djs66256.short_drama.feature.earn.model

data class EarnTaskContext(
    val taskId: String,
    val source: String = EARN_SOURCE,
    val returnTarget: String = EARN_RETURN_TARGET,
    val videoId: String,
) {
    fun isValid(): Boolean {
        return taskId.isNotBlank() &&
            source == EARN_SOURCE &&
            returnTarget == EARN_RETURN_TARGET &&
            videoId.isNotBlank()
    }
}

File: ios/ShortDrama/Sources/Domain/Entities/EarnTaskContext.swift
import Foundation

struct EarnTaskContext: Hashable, Identifiable, Sendable {
    let taskId: String
    let source: String
    let returnTarget: String
    let videoId: String

    var id: String {
        "\(taskId)-\(videoId)-\(returnTarget)"
    }

    init?(
        taskId: String,
        source: String,
        returnTarget: String,
        videoId: String
    ) {
        let normalizedTaskId = taskId.trimmingCharacters(in: .whitespacesAndNewlines)
        let normalizedSource = source.trimmingCharacters(in: .whitespacesAndNewlines)
        let normalizedReturnTarget = returnTarget.trimmingCharacters(in: .whitespacesAndNewlines)
        let normalizedVideoId = videoId.trimmingCharacters(in: .whitespacesAndNewlines)

        guard !normalizedTaskId.isEmpty,
              normalizedSource == "earn",
              normalizedReturnTarget == "/earn",
              !normalizedVideoId.isEmpty else {
            return nil
        }

        self.taskId = normalizedTaskId
        self.source = normalizedSource
        self.returnTarget = normalizedReturnTarget
        self.videoId = normalizedVideoId
    }
}
